AsyncDisplayKit/ASImageNode.mm

```diff
@@ -111,7 +111,9 @@ - (instancetype)initWithViewBlock:(ASDisplayNodeViewBlock)viewBlock
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
   ASDN::MutexLocker l(_imageLock);
-  if (_image)
+  if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO)
+    return CGSizeMake(MIN(constrainedSize.width, self.preferredFrameSize.width), MIN(constrainedSize.height, self.preferredFrameSize.height));
```

Comment:
Hmmm...I worry this may not work for the original use case I had. If we have an image node that we set the preferredFrameSize to be 200x200 but we have a local image that is 400x400, the image view will never respect our preferredFrameSize.

What if we flip the logic:

```
- (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
{
  ASDN::MutexLocker l(_imageLock);
  if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO)
    return [super calculateSizeThatFits:constrainedSize];

  return _image.size;
}
```

Depending on how network images are cached the same thing could happen.